### PR TITLE
Automate release with JReleaser

### LINKED ISSUE #3602: Automate releases using JReleaser
The idea is to automate the releases so that we are able to release faster with a more regular schedule

### PR DESCRIPTION
**Pull request by filiphr**
Add JReleaser for automating the release and add a step for automating the publishing of the website

This has been heavily inspired by the [Moditect](https://github.com/moditect/moditect/blob/main/.github/workflows/release.yml) and [JReleaser](https://github.com/jreleaser/jreleaser/blob/main/.github/workflows/release.yml) release procedures.

Not sure if @aalmiray, the creator of JReleaser, might be able to chime in and let us know if I'm doing something incorrectly here.  Not sure what the best approach would be to test this with an actual Maven release. 

I think doing the 1.6.0.Beta2 release would be a good first try for this.

Closes #3602

### CODE REVIEW

**Comment by aalmiray** [comment]
FWIW you can test most of the release configuration locally. You may run `mvn jreleaser:jreleaserConfig` to print out the resolved configuration. You may also run a release in dry run mode. 

More tips can be found at https://jreleaser.org/guide/latest/checklist.html

File: parent/pom.xml [hunk 1]
```diff
@@ -823,18 +823,93 @@
...
+                                                </artifactOverrides>
+                                            </maven-central>
+                                        </nexus2>
+                                    </maven>
+                                </deploy>
+                                <release>
+                                    <github>
+                                        <tagName>{{projectVersion}}</tagName>
+                                        <releaseName>{{projectVersion}}</releaseName>
+                                        <overwrite>false</overwrite>
```
Comment (aalmiray) [inline parent/pom.xml:891]: This is the default setting and thus may be omitted.

File: parent/pom.xml [hunk 2]
```diff
@@ -823,18 +823,93 @@
...
+                                        <update>
```
Comment (aalmiray) [inline parent/pom.xml:892]: This setting is needed when you would like to update an existing release by modifying: title, release notes, and/or release assets.

File: parent/pom.xml [hunk 3]
```diff
@@ -823,18 +823,93 @@
...
+                                            <enabled>true</enabled>
+                                        </update>
+                                        <changelog>
+                                            <enabled>true</enabled>
+                                            <external>${maven.multiModuleProjectDirectory}/NEXT_RELEASE_CHANGELOG.md</external>
+                                        </changelog>
+                                    </github>
+                                </release>
+                                <files>
```
Comment (aalmiray) [inline parent/pom.xml:901]: Alternatively, if these artifacts match a BINARY or JAVA_BINARY distribution then consider using the `<distributions>` section https://jreleaser.org/guide/latest/concepts/distributions/index.html

File: parent/pom.xml (hunk 2 above)
Comment (filiphr) [reply parent/pom.xml:892]: Good point, I think that I wanted to set the `overwrite` to `true`. However, I might have found a better approach by using the external makrdown.

File: parent/pom.xml (hunk 3 above)
Comment (filiphr) [reply parent/pom.xml:901]: Thanks for the pointer. I was reading up on the binaries, but I couldn't really understand if they are a right fit for this or not. This is non an executable binary, but a zip containing the sources, releases JARs, the Javadoc and the documentation

**Comment by filiphr** [comment]
Thanks a lot for the comments @aalmiray. I was already trying it with the dry run locally and it looked like it is working correctly. 

Thanks a lot for your work on JReleaser, I think automating the release is going to make us release faster 😄

File: parent/pom.xml (hunk 3 above)
Comment (aalmiray) [reply parent/pom.xml:894]: Ah gotcha, then using the `files` section is the way to go.

**Pull request merged**